From 06ade048c171160379dd906ed1efabb90e5d5ed4 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:36:54 -0400
Subject: [virt] kvm: use a more sensible error number when debugfs directory creation fails

Message-id: <1406040016-3289-20-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86098
O-Subject: [RHEL7 PATCH v2 019/221] KVM: use a more sensible error number when debugfs directory creation fails
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

I don't know if this was due to cut and paste, or somebody was really
using a D20 to pick the error code for kvm_init_debugfs as suggested by
Linus (EFAULT is 14, so the possibility cannot be entirely ruled out).

In any case, this patch fixes it.

Reported-by: Tim Gardner <tim.gardner@canonical.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit 0c8eb04a6241da28deb108181213b791c378123b)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/virt/kvm/kvm_main.c b/virt/kvm/kvm_main.c
index 21e9cc0..becb56d 100644
--- a/virt/kvm/kvm_main.c
+++ b/virt/kvm/kvm_main.c
@@ -3110,7 +3110,7 @@ static const struct file_operations *stat_fops[] = {
 
 static int kvm_init_debug(void)
 {
- int r = -EFAULT;
+ int r = -EEXIST;
  struct kvm_stats_debugfs_item *p;
 
  kvm_debugfs_dir = debugfs_create_dir("kvm", NULL);
-- 
1.7.1